### ISSUE TITLE
Add anonymised interaction collection to monitor wallet use

### ISSUE THREAD

**Issue by mvaivre** [alephium/desktop-wallet#477]
**First use case : collect how many addresses our users have.**

Other potential use cases : 

- At what frequency is the wallet opened?
- How frequently is the main address changed?
- How many times are the advanced actions used?
- How frequently is the scan to import feature used?
- How many wallets?
- ...

**Comment by nop33** [comment]
## What is Ledger Live using

Ledger Live seems to be using [Segment](https://segment.com/) for analytics. I couldn't, however, see any related npm packages. I dived into the source code and found these files related to analytics:
- https://github.com/LedgerHQ/ledger-live/tree/develop/apps/ledger-live-desktop/src/renderer/analytics

In [`src/renderer/index.html`](https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-desktop/src/renderer/index.html):
```html    
    <script
      async
      src="https://resources.live.ledger.app/public_resources/analytics.min.js"
    ></script>
```

## Interesting resources
- Blog [post](https://kilianvalkhof.com/2018/apps/using-google-analytics-to-gather-usage-statistics-in-electron/) on adding Google Analytics in Electron apps (using an npm package called [`universal-analytics`](https://www.npmjs.com/package/universal-analytics) with using an anonymized ID for identifying the user.

**Comment by mvaivre** [comment]
Two thoughts:
- How needs are pretty basic IMO.
- We want to be transparent about what we record and how/where we record.

2 solutions: either use an existing self-hosted analytics solution, or create our own simple service (simple DB + REST or GraphQL server).

Self hosted solution:

- **[Plausible](https://github.com/plausible/docs)** seems to be the most popular choice, very simple but it's not really customisable according to my quick research. It's mostly focused on page views. We could use "events" though. Easy to setup (docker).
- **[Umami](https://github.com/umami-software/umami)** is a similar simple solution, with the same flaws IMO.
- **[Posthog](https://github.com/posthog/posthog)** looks pretty nice, dev oriented, event based. Their events + actions seem pretty customisable. It also provides a React Native lib! Based on my quick research, this would be my preferred library.
- Other solutions, not customisable enough IMO, too "website oriented", too feature rich: Matomo, Offen, Countly...


TLDR: we either deploy a Posthog stack and use their actions related endpoints, or we quickly develop our own service without wasting too much time on DB schema building etc. as we'd be the only user consuming data.

**Comment by nop33** [comment]
Posthog looks like a great proposal! ⭐⭐⭐⭐⭐

## Implementing use cases

When the app is launched, we can check if `analytics_user_id` is stored in localStorage. If it's not, we create one and store it:

```js
const analyticsUserId = nanoid()
localStorage.setItem('analytics_user_id', analyticsUserId)
```

Then, we identify the user to posthog (or we can pass the id with every event capture, need to look more into this):

```js
posthog.identify(analyticsUserId)
```

### At what frequency is the wallet opened?

We could trigger an event when the app is launched:

```js
posthog.capture('app lauched')
```

I guess no need to pass a property `timestamp`, since the event must be stored with some sort of timestamp itself. Otherwise we could do:

```js
posthog.capture('app lauched', { timestamp: new Date() })
```

### How frequently is the default address changed?

We can trigger an event once the address is changed from:
- Address creation form (set to default)
- Clicking the "ribbon" button from the list of addresses

```js
posthog.capture('default address changed')
```

### How many times are the advanced actions used?

We could trigger an event when the "Submit" button of each of the modals of the advanced operations is clicked:

```js
posthog.capture('UTXOs consolidated')
posthog.capture('1 address per group generated')
posthog.capture('Addresses discovered manually')
```

### How frequently is the scan-to-import feature used?

We could trigger an event once the camera captures the QR code and the import was successful

```js
posthog.capture('Imported wallet via QR code')
```

### How many wallets?

I think that's a bit tricky cause it's not an event/action. Maybe it would make sense to just capture the "Created/Imported new wallet" event. Since all events of a particular user will be associated with them through a unique ID that is stored in localStorage on the first app launch after installation, we can calculate the number of wallets based on this metric (maybe also add a metric for deleting a wallet).

```js
posthog.capture('Created new wallet')
posthog.capture('Imported wallet')
posthog.capture('Deleted wallet')
```

Another approach would be to set a user property. On app launch, we could include the number of wallets the user has, using the [`$set` property](https://posthog.com/docs/integrate/client/js#set):

```js
posthog.capture('app lauched', { $set: { wallets: number_of_wallets } })
```

## Other cool things to track

I think we should ask also the rest of the team, but here's some ideas:

Events to track:
- Devtools enabled
- Deployed smart contract
- Called smart contract
- Created contact
- Changed network settings

User properties to register:
- Theme used
- Prefered wallet lock time
- Prefered password requirement before sending txs
- Language

## Things to keep in mind

There's the `autocapture` setting that it's by [default on](https://posthog.com/docs/integrate/client/js). It tracks many things out of the gate, like clicks on buttons, pageviews, etc. I would say we turn it off so that we collect only data that we specifically want. Another thing is that [session recording](https://posthog.com/manual/recordings) is on by default. I think we should turn that off as well

```
autocapture: false
disable_session_recording: true
```

**Comment by mvaivre** [comment]
Great implementation examples! This looks good! 💯 
Also fully aligned with the other cool things to track. 
Things to keep in mind: 💯% aligned as well. Let's focus solely on custom events for now.

**Comment by nop33** [comment]
Closing since it's merged in v2.0!
